### ISSUE TITLE
Unknown FFmpeg error

### ISSUE THREAD

**Issue by aleksamagicka** [snobu/destreamer#82]
I try to download a Stream video, and it goes to 100% without errors, but then crashes with `Unknown FFmpeg error`. Is there anything I can do to debug this?

`ffmpeg` crashed for me even before the tokencache update, but at least it left the files so I could manually join audio and video. Thank you for any help.

**Comment by snobu** [comment]
Try ffmpeg 4.2.2, that's what we heavily tested with. Maybe there won't be a need to debug after all.

**Comment by aleksamagicka** [comment]
I'm on ffmpeg 4.2.2 already. Tried with the latest one, the results are the same. Here is the whole output:

```
Using ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers

Username: ...
Video URLs:
[
  'https://web.microsoftstream.com/video/d604a5cc-c15c-4aff-918b-a529eb0da92a'
]

Access token has expired.
Launching headless Chrome to perform the OpenID Connect dance...
Navigating to login page...
We are logged in.
Wrote access token to token cache.
At this point Chromium's job is done, shutting it down...

Fresh access token dropped into .token_cache
Video GUIDs:
[ 'd604a5cc-c15c-4aff-918b-a529eb0da92a' ]
Fetching metadata...
Calling https://euwe-1.api.microsoftstream.com/api/videos/d604a5cc-c15c-4aff-918
b-a529eb0da92a?api-version=1.4-private
[ 'videos' ]

Downloading Video: Meeting in "?????????"

Spawning ffmpeg with access token and HLS URL. This may take a few seconds...
Using a fresh cookie.
progress [██████████████████████████] 100% 188.8kbits/s 1s

ffmpeg returned an error: [aac @ 000000000590ed40] Qavg: 3503.457



Unknown FFmpeg error
```

**Comment by snobu** [comment]
Try changing this line to `console.log(error);` -
https://github.com/snobu/destreamer/blob/master/src/destreamer.ts#L232

Rebuild and try again.

That should give you the full ffmpeg output, i hope.

**Comment by aleksamagicka** [comment]
Thanks for the help. I solved the problem (for now, tested it on a few videos) by recreating the `videos` folder. Closing this for now, will report back if it occurs again.

On a side note, what do you think about using `console.dir(error)` when `-v` is passed, instead of simply saying `Unknown FFmpeg error`? Before reporting this, while I tested, I thought that verbose mode would give me more info about the error.

**Comment by snobu** [comment]
Yes, we will add that, it's a good suggestion, thanks.

**Comment by plopresti** [comment]
@aleksamagicka  I have this issue but I do not understand what the problem is nor what you did to fix it... Also "this line" no longer seems to refer to anything useful.

Help?

**Comment by plopresti** [comment]
Solved it. Different problem for me I think.

Running with environment variable "FFREPORT=file=ffreport.log" showed that my ffmpeg (version 5.1.3) was complaining about "error while loading shared libraries: libopenh264.so.7". So I downloaded libopenh264 from https://github.com/cisco/openh264/releases/tag/v2.3.1, uncompressed it, symlinked libopenh264.so.7 -> libopenh264.2.3.1-linux64.7.so, added the directory to LD_LIBRARY_PATH...

...and now my destreamer is running! Thank you for this tool.
